back/engines/commercial/analytics/app/services/analytics/matomo_data_importer.rb

```diff
@@ -53,7 +53,14 @@ def persist_visit_data(visit_data)
     def update_date_dimensions(visits)
       from = timestamp_to_date(visits.pluck('firstActionTimestamp').min)
       to = timestamp_to_date(visits.pluck('lastActionTimestamp').max)
-      Analytics::PopulateDimensionsService.create_dates(from, to)
+      (from..to).each do |date|
+        Analytics::DimensionDate.create_or_find_by(
```

Comment:
Changed this (as suggested before) as I wondered whether the bang version of create in PopulateDimensionsService was causing problems

Reply:
Probably, on the DB level it's the same. [The documentation](https://apidock.com/rails/v6.0.0/ActiveRecord/Relation/create_or_find_by) says

> the exception such an insertion would normally raise is caught

Maybe `find_or_create_by` would behave differently, but I'm not sure.